tools/stress-cluster/cluster/provision.ps1

```diff
@@ -1,32 +1,84 @@
+[CmdletBinding(SupportsShouldProcess = $true)]
 param (
     [string]$Environment = 'dev',
     [string]$Namespace = 'stress-infra',
-    [switch]$Development = $false
+    [switch]$Development = $false,
+
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $TenantId,
+
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+    [string] $ProvisionerApplicationId,
+
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [string] $ProvisionerApplicationSecret,
+
+    # Enables passing full json credential config without throwing unrecognized parameter errors
+    [Parameter(ValueFromRemainingArguments = $true)]
+    $RemainingArguments
 )
 
+$ErrorActionPreference = 'Stop'
+$LastWhatIfPreference = $WhatIfPreference
+$WhatIfPreference = $false
+. (Join-Path $PSScriptRoot "../../../eng/common/scripts/Helpers" PSModule-Helpers.ps1)
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+$WhatIfPreference = $LastWhatIfPreference
```

Comment:
Don't quite understand why we are changing WhatIfPreference to false and then to original value again

Reply:
It's somewhat annoying. There are sub-components called within `Install-ModuleIfNotInstalled` that don't run when `-WhatIf` is set that we absolutely need to run, but because the top level cmdlet doesn't have the `SupportsShouldProcess` binding, I don't think I can just pass `-WhatIf:$false`. We need this dependency installed whether we are in whatif mode or not.

Reply:
Same for `Import-Module`. It doesn't support `-WhatIf` but if `Install-ModuleIfNotInstalled` doesn't return a module, then we throw.

Reply:
I updated the underlying function to add `SupportsShouldProcess`, so it should work now.